### PR TITLE
Workload Identity: allow passing roles to created service account

### LINKED ISSUE #694: Specify roles that should be added to Service Account created for Workload identity
```hcl
module "workload_identity" {
  source       = "terraform-google-modules/kubernetes-engine/google//modules/workload-identity"
  version      = "~> 11.1"
  name         = "prod-sa"
  namespace    = "default"
  cluster_name = module.gke.name
  project_id   = var.project_id
  roles        = ["roles/editor", "roles/dns.admin"]
}
```
Something like this

### PR DESCRIPTION
**Pull request by MeNsaaH**
Closes #694

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/terraform-google-kubernetes-engine/modules/workload-identity/README.md /tmp/tmp.2TE1kt0tON/generate_docs/workspace/terraform-google-kubernetes-engine/modules/workload-identity/README.md
78a79
> | roles | (optional) A list of roles to be added to the created Service account | list(string) | `<list>` | no |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ```

File: modules/workload-identity/main.tf [hunk 1]
```diff
@@ -64,3 +64,12 @@ resource "google_service_account_iam_member" "main" {
   role               = "roles/iam.workloadIdentityUser"
   member             = local.k8s_sa_gcp_derived_name
 }
+
+
+resource "google_project_iam_member" "workload_identity_sa_bindings" {
+  count = length(var.roles)
```
Comment (bharathkkb) [inline modules/workload-identity/main.tf:70]: Can we use `for_each`?

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: modules/workload-identity/main.tf (hunk 1 above)
Comment (MeNsaaH) [reply modules/workload-identity/main.tf:70]: for each works with maps if I'm not mistaken. I'll need to verify that

Comment (MeNsaaH) [reply modules/workload-identity/main.tf]: Done

**Review by bharathkkb — APPROVED** [review]
LGTM
/cc @morgante

**Comment by morgante** [comment]
Thanks for the contribution!

**Pull request merged**